torchmultimodal/models/mdetr.py

```diff
@@ -0,0 +1,699 @@
+            dropout (float): Dropout value. Default: 0.1.
+            activation (Callable[..., Tensor]): The activation function of the
+                intermediate layer. Default: relu
+            normalize_before (bool): Whether to do PreNorm. Default: False
+            return_intermediate (bool): Whether to return intermediate decoder outputs.
+                Default: False
+            pass_pos_and_query (bool): Whether to pass positional encodings to
+                attention layers. Default: True
+
+
+    Inputs: src (Optional[Tensor]): The image input. Default: None
+            mask (Optional[Tensor]) The mask for the image sequence. Default: None
+            query_embed (Optional[Tensor]): Positional embeddings applied to Q
+                cross-attention matrix in decoder. Default: None
+            pos_embed (Optional[Tensor]): Positional embeddings applied to Q and K
+                self-attention matrices in decoder. Default: None
+            query_pos (Optional[Tensor]): Positional embeddings applied to Q
+                cross-attention matrix. Default: None
+            text_memory (Optional[Tensor]): Text input. Default: None
+            text_attention_mask (Optional[Tensor]): Attention mask for text input.
+                Default: None
+
+    """
+
+    def __init__(
+        self,
+        d_model: int = 512,
+        nhead: int = 8,
+        num_encoder_layers: int = 6,
+        num_decoder_layers: int = 6,
+        dim_feedforward: int = 2048,
+        dropout: float = 0.1,
+        activation: Callable[..., Tensor] = nn.functional.relu,
+        normalize_before: bool = False,
+        return_intermediate_dec: bool = False,
+        pass_pos_and_query: bool = True,
+    ):
+        super().__init__()
+
+        self.pass_pos_and_query = pass_pos_and_query
+        encoder_layer = TransformerEncoderLayer(
+            d_model, nhead, dim_feedforward, dropout, activation, normalize_before
+        )
+        encoder_norm = nn.LayerNorm(d_model) if normalize_before else None
+        self.encoder = TransformerEncoder(
+            encoder_layer, num_encoder_layers, encoder_norm
+        )
+
+        decoder_layer = TransformerDecoderLayer(
+            d_model, nhead, dim_feedforward, dropout, activation
+        )
+        decoder_norm = nn.LayerNorm(d_model)
+        self.decoder = TransformerDecoder(
+            decoder_layer,
+            num_decoder_layers,
+            decoder_norm,
+            return_intermediate=return_intermediate_dec,
+        )
+        self.d_model = d_model
+        self._reset_parameters()
+
+    def _reset_parameters(self):
+        for p in self.parameters():
+            if p.dim() > 1:
+                nn.init.xavier_uniform_(p)
+
+    def forward(
+        self,
+        src: Optional[Tensor] = None,
+        mask: Optional[Tensor] = None,
+        query_embed: Optional[Tensor] = None,
+        pos_embed: Optional[Tensor] = None,
+        text_memory: Optional[Tensor] = None,
```

Comment:
You're right, it's probably not optional. Before almost everything was optional because they called transformer twice (once to encode, once to decode), so need to update these